### PR TITLE
SW-4941 In-app notifications for deliverable ready for review and status updated

### PR DESCRIPTION
**Pull request by karthikbtf**
- store in-app notifications for accelerator team and tf contact, if deliverable ready for review
- store in-app notification for org admin, owner, manager if deliverable status updated
- moved getTerraformationContact from EmailNotificationService into UserStore since this is now used in both notification services
- added tests

### CODE REVIEW

File: src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt [hunk 1]
```diff
@@ -349,6 +388,26 @@ class AppNotificationService(
...
+  private fun insertAcceleratorNotification(
+      localUrl: URI,
+      notificationType: NotificationType,
+      organizationId: OrganizationId,
+      renderMessage: () -> NotificationMessage,
+  ) {
+    val recipients = HashSet(userStore.fetchWithGlobalRoles())
+    val tfContact = userStore.getTerraformationContactUser(organizationId)
+
+    recipients.apply { if (tfContact != null) add(tfContact) }
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt]: Is there a functional different to writing it like this?
```
if (tfContact != null) recipients.add(tfContact);
```

File: src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt [hunk 2]
```diff
@@ -349,6 +388,26 @@ class AppNotificationService(
     }
   }
 
+  private fun insertAcceleratorNotification(
+      localUrl: URI,
+      notificationType: NotificationType,
+      organizationId: OrganizationId,
+      renderMessage: () -> NotificationMessage,
+  ) {
+    val recipients = HashSet(userStore.fetchWithGlobalRoles())
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt:397]: Should this be scoped to TF experts only?

File: src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt (hunk 1 above)
Comment (karthikbtf) [reply src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt]: Not sure why I did it this way. I can change it.

File: src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt (hunk 2 above)
Comment (karthikbtf) [reply src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt:397]: By TF Experts they mean all the current global roles, had clarified with Leann.
Not specifically the TF Expert role.

Comment (nickgraz) [reply src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt:397]: Ah, I see, thanks.

**Review by nickgraz — APPROVED** [review]
Looks good to me!

File: src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt [hunk 3]
```diff
@@ -283,6 +287,41 @@ class AppNotificationService(
...
+    // This is run as a system user because the org membership permission checks don't apply
+    // here. The recipient of the notification may not be a member (TF contact) in the participant
+    // org.
+    systemUser.run {
+      val participant = participantStore.fetchOneById(event.participantId)
+      val deliverableUrl = webAppUrls.acceleratorConsoleDeliverable(event.deliverableId)
+      val renderMessage = { messages.deliverableReadyForReview(participant.name) }
+
+      log.info(
+          "Creating app notifications for participant ${participant.name}'s deliverable ${event.deliverableId} ready for review")
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt]: Maybe log the participant ID here too?

**Review by sgrimm — APPROVED** [review]

**Pull request merged**